lib/Tasks/Sending.php

```diff
@@ -97,11 +98,22 @@ public static function createManyFromTasks($tasks) {
     foreach ($tasks as $task) {
       if (!empty($queuesIndex[$task->id])) {
         $result[] = self::create($task, $queuesIndex[$task->id]);
+      } else {
+        static::handleInvalidTask($task);
       }
     }
     return $result;
   }
 
+  public static function handleInvalidTask(ScheduledTask $task) {
+    $loggerFactory = LoggerFactory::getInstance();
+    $loggerFactory->getLogger(LoggerFactory::TOPIC_NEWSLETTERS)->addError(
+      'invalid sending task found, deleting',
+      ['task_id' => $task->id]
+    );
+    $task->delete();
```

Comment:
I have a concern about deleting the task. When we delete it we also delete the task subscribers' data. The original idea was to add a new state `invalid`. If someone reports an issue with a stuck newsletter we would be able to recreate the sending queue and we may have more data for debugging. On the other hand, we can't easily match a sending task with a newsletter so maybe it is ok to delete it. Maybe keeping more data is a safer option. I don't have a strong opinion here. Just wanted to point this out. What do you think?

Reply:
I chose to delete tasks without sending queue only because we always did the same with tasks without newsletters. If you think keeping them as `invalid` would help in debugging, let's do that. I've made the change.